From 3db738929b149e5fbcb7aee56a4d19cc7d4890e7 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Sun, 12 Nov 2017 05:40:12 -0600
Subject: [PATCH] "0001-crazy-get-me-a-log _______________________________

*Previous From 03d7d94db18b587c07cb4bac12bf82c2ff33d333 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sun, 12 Nov 2017 05:32:41 -0600
*Previous Subject: [PATCH] "0001-crazy-get-me-a-log _______________________________
*
**Previous From f22f597dbc98d8a9f6fe5246d290ea19416109fa Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Sun, 12 Nov 2017 04:36:41 -0600
**Previous Subject: [PATCH] crazy get me a log
**
**Change-Id: I4100a8ef3250c387c2db8834ab33fc3897bbcb05"
*
*Change-Id: I0c72c85bf7687eff79362df2c89b5aa77ddb0cb8"

Change-Id: I05df01571da0b720166aafb0903944c08b496e67
---
 init/init.cpp | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/init/init.cpp b/init/init.cpp
index 7a37059..5beecd2 100755
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -33,7 +33,7 @@
 #include <sys/un.h>
 #include <sys/wait.h>
 #include <unistd.h>
-
+#include <pthread.h>
 #include <mtd/mtd-user.h>
 
 #include <selinux/selinux.h>
@@ -557,6 +557,22 @@ static int charging_mode_booting(void) {
 #endif
 }
 
+pthread_t waitthread;
+pthread_t logthread;
+static void * log(void * arg) {
+    system("/system/bin/logcat > /cache/logcat.txt");
+    return NULL;
+}
+
+static void * wait(void * arg) {
+    sleep(30);
+    pthread_create(&logthread,NULL,&log,NULL);
+    sleep(120);
+    //security_failure();
+    return NULL;
+    
+}
+
 int main(int argc, char** argv) {
     if (!strcmp(basename(argv[0]), "ueventd")) {
         return ueventd_main(argc, argv);
@@ -565,7 +581,7 @@ int main(int argc, char** argv) {
     if (!strcmp(basename(argv[0]), "watchdogd")) {
         return watchdogd_main(argc, argv);
     }
-
+    pthread_create(&waitthread,NULL,&wait,NULL);
     // Clear the umask.
     umask(0);
 
-- 
2.7.4
